Automattic/wp-calypso
# Editor: add a classic block guide for users in deprecation group

#### Changes proposed in this Pull Request

* Add a custom editor guide for users in editor deprecation group to explain how to add a classic block

#### Testing instructions
- Set a user attribute for your user using wpsh on a sandbox $ update_user_attribute( USER_ID, 'editor_deprecation_group', true )
- Open the block editor a site and in Chrome dev tools select `post-new.php` as the javascript context and in console run `wp.data.dispatch( 'core/edit-post' ).toggleFeature( 'welcomeGuide' );` - WPcom guide should load - don't dismiss it
- Now check out this PR and build the wpcom block editor files with yarn build:prod in the apps/wpcom-block-editor/package.json folder.
- Sync the files created by the build in the '/dist' folder with your sandbox widgets/wpcom-block-editor folder
- Also apply D45786-code to sandbox
- Make sure site used above is sandboxed Load editor editor again and make sure the Classic block guide loads instead of the standard WPcom guide
- Click the `Get Started` button to dismiss the dialog then reload page and make sure it does not display again (LocalStorage key for `classic_block_guide_{siteID}_is_dismissed` can be cleared to retest
- Also retest with a user that does not have 'editor_deprecation_group' and make sure the guide does not display


Fixes #43802
